### PR TITLE
Improve Stack sample code for performance

### LINKED ISSUE #9385: Performance: Stack.Horizontal.Wrap.Example.tsx has bad CSS example lead to bad browser performance.
### Environment Information 
 
- __Package version(s)__: (fill this out) 
- __Browser and OS versions__: (fill this out if relevant)

#### Actual behavior:
I am Code Reviewing peer's change and flag the browser render engine (Edge, Blink, Trident) potential rendering engine over head,  He point out he copy that from the Sample code. 

https://github.com/OfficeDev/office-ui-fabric-react/blob/1b705beb1cad712f93da2954511b7d6a1d583797/packages/office-ui-fabric-react/src/components/Stack/examples/Stack.Horizontal.Wrap.Example.tsx#L22

This is out of experience from working with Edge perf lead.
In general, CSS should avoid &lt;tag&gt; name. It is especially worst when it is last of the CSS rule.
1. Browser tend to optimize for div/span with class name (Hash)
2.  Browser tend eval the rule to apply form right to left. 
      --- when you specified rule end with &lt;span&gt;  that lead to Hash look up failed, end up gather every &lt;span&gt; on the page!! then leading down to bad browser performance.

This example happen to hit both.  In general, example should avoid performance pitfall.
 

 
#### Expected behavior:
Change to the last &lt;span&gt; a class name 
 
https://github.com/OfficeDev/office-ui-fabric-react/blob/1b705beb1cad712f93da2954511b7d6a1d583797/packages/office-ui-fabric-react/src/components/Stack/examples/Stack.Horizontal.Wrap.Example.tsx#L22

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (Yes, No)
 
Requested priority: (Blocking, High, Normal, Low) 
 
Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by Rajeev-K**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #9385 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

CSS performance is improved when the rightmost selector in a compound selector matches the fewest number of elements. CSS class names, as opposed to tag names are therefore preferred as the last selector. For more info see:
https://www.stevesouders.com/blog/2009/06/18/simplifying-css-selectors/
https://stackoverflow.com/questions/5797014/why-do-browsers-match-css-selectors-from-right-to-left

#### Focus areas to test

Sample code for Stack control

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9403)

### CODE REVIEW

**Comment by khmakoto** [comment]
@dzearing I can approve, just wanted to double check with you if this is a pattern we want to be promoting in the examples.

File: packages/office-ui-fabric-react/src/components/Stack/examples/Stack.Horizontal.HorizontalAlign.Example.tsx [hunk 1]
```diff
@@ -5,18 +5,17 @@ export class HorizontalStackHorizontalAlignExample extends React.Component<{}, {
   public render(): JSX.Element {
     const styles = mergeStyleSets({
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Stack/examples/Stack.Horizontal.HorizontalAlign.Example.tsx]: move this out of render. It's not using any properties.

File: packages/office-ui-fabric-react/src/components/Stack/examples/Stack.Horizontal.WrapAdvanced.Example.tsx [hunk 2]
```diff
@@ -32,18 +32,17 @@ export class HorizontalStackWrapAdvancedExample extends React.Component<{}, IExa
       root: {
         background: DefaultPalette.themeTertiary,
         width: `${stackWidth}%`,
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Stack/examples/Stack.Horizontal.WrapAdvanced.Example.tsx]: @khmakoto this here is an anti-pattern. You're going to mutate the width of a thing, which will cause re-evaluation of the styles and a re-calculation of classnames. A better approach is to use inline styling for constantly mutating widths or positions. That way you keep the classname generation minimized to static content.

**Comment by dzearing** [comment]
@khmakoto I left some comments here. I think the styles should be calculated outside of the render. It's fine to use the subitem class instead of a selector. It's not OK to inject frequently mutated values (sliders) to the styles, but instead to use inline styling.

I think this PR would be fine to merge, if you want to follow up on the other items.

**Comment by khmakoto** [comment]
@dzearing sounds good! I'll merge this PR and follow up on your comments on another PR. Thanks for the review!!

**Pull request merged**